src/python.cpp

```diff
@@ -995,6 +995,21 @@ PyObject* r_to_py(RObject x, bool convert) {
            "converted");
     }
 
+    int flags = NPY_ARRAY_FARRAY_RO;
+
+    // because R logical vectors are just ints under the
+    // hood, we need to explicitly construct a boolean
+    // vector for our Python array. note that the created
+    // array will own the data so we do not free it after
+    if (typenum == NPY_BOOL) {
+      R_xlen_t n = XLENGTH(sexp);
+      bool* converted = (bool*) std::malloc(n * sizeof(bool));
```

Comment:
Is there a numpy memory allocation function or macro? (just to be 100% sure that the same heap is used for malloc and free)

Reply:
You're right; it looks like there's a `PyArray_malloc()` function (https://docs.scipy.org/doc/numpy-1.7.0/reference/c-api.array.html#memory-management). I'll update the PR.

Reply:
Reading a bit here:

https://github.com/numpy/numpy/blob/f5758d6fe15c2b506290bfc5379a10027617b331/numpy/core/include/numpy/ndarraytypes.h#L348-L367

It looks like NumPy explicitly uses the 'raw' allocation APIs in Python. That is, while it is possible to register custom allocators in Python (appears to be new in Python 3.6), NumPy explicitly requests the 'raw' allocators when possible. Note that these effectively just delegate to `malloc()`:

https://github.com/python/cpython/blob/28f713601d3ec80820e842dcb25a234093f1ff18/Objects/obmalloc.c#L66-L76

I think this implies the code here should be good 'as-is' since all the indirection in NumPy is around making sure the raw `malloc()` routine is used.